pkg/endpoint/endpoint.go

```diff
@@ -851,8 +851,17 @@ func parseEndpoint(ctx context.Context, owner regeneration.Owner, bEp []byte) (*
 	// If host label is present, it's the host endpoint.
 	ep.isHost = ep.HasLabels(labels.LabelHost)
 
-	// Overwrite datapath configuration with the current agent configuration.
-	ep.DatapathConfiguration = NewDatapathConfiguration()
+	// Only overwrite datapath configuration related with enable-endpoint-routes
```

Comment:
It would be helpful to state why in this comment as well

Reply:
I try to add some explannations and rebase the commit. But the pull request seems not updated. So I make a new pull requset 16289